File: Java/프로그래머스/2/42587. 프로세스/프로세스.java
import java.util.*;

class Solution {
    public int solution(int[] priorities, int location) {
        int answer, cur_idx, max_idx;
        Deque<Integer> queue = new ArrayDeque<>();
        List<Integer> pos = new ArrayList<>();
        
        for (int i=0; i<priorities.length; i++) queue.add(i);
        
        while(!queue.isEmpty()) {
            cur_idx = queue.pop();
            max_idx = Arrays.stream(priorities).max().getAsInt();
            
            if (priorities[cur_idx] < max_idx) queue.add(cur_idx);
            else {
                priorities[cur_idx] = -1;
                pos.add(cur_idx);
            }
        }
        answer = pos.indexOf(location) + 1;
        return answer;
    }
}

File: JavaScript/프로그래머스/2/42587. 프로세스/프로세스.js
function solution(priorities, location) {
    var queue = [];
    var pos = [];
    var cur_idx, max_idx, answer;
    
    for (let i=0; i<priorities.length; i++) queue.push(i);

    while(queue.length > 0) {
        cur_idx = queue.shift();
        max_idx = Math.max(...priorities);
        
        if (priorities[cur_idx] < max_idx) queue.push(cur_idx);
        else {
            pos.push(cur_idx);
            priorities[cur_idx] = -1;
        }
    }
    answer = pos.indexOf(location) + 1;
    return answer;
}